### PR TITLE
docs(error-reporting): add reporting client errors recipe

### PR DESCRIPTION
**Pull request by JAdshead**
## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,8 +29,8 @@ Want to get paid for your contributions to `one-app`?
 
 * [Features](#-features)
 * [Usage](#-usage)
-* [Recipes](#-recipes)
-* [API](#%EF%B8%8F-api)
+* [Recipe Docs](./docs/recipes/README.md)
```
Comment (JAdshead) [inline README.md:32]: having the additional click was very irritating

File: docs/recipes/Reporting-Client-Errors.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+ErrorBoundary.propTypes = {
+  children: PropTypes.node,
+  dispatch: PropTypes.func.isRequired,
+};
+
+export default connect()(ErrorBoundary);
+```
+
+Read more about [error boundaries](https://reactjs.org/docs/error-boundaries.html) the React website.
```
Comment (Francois-Esquire) [inline docs/recipes/Reporting-Client-Errors.md:88]:
```suggestion
Read more about [error boundaries](https://reactjs.org/docs/error-boundaries.html) on the React website.
```

File: docs/recipes/Reporting-Client-Errors.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+# Reporting Client Errors
+
+## `errorReporting` Duck
+
+`one-app` includes the [`errorReporting`](https://github.com/americanexpress/one-app-ducks#errorreporting-duck)
+duck from `one-app-ducks` which will log any errors reported by dispatching the [`addErrorToReport`](https://github.com/americanexpress/one-app-ducks#adderrortoreport) action. This provides your modules with a simple and efficient way to report errors from the client.
+
+When `addErrorToReport` is dispatched during SSR the `errorReporting` duck will log the reported error
+to `console.error`. If dispatched on the client `addErrorToReport` will result in the error being `POST`ed
+to the `reportingUrl` configured by the [`ONE_CLIENT_REPORTING_URL`](https://github.com/americanexpress/one-app/blob/main/docs/api/server/Environment-Variables.md#one_client_reporting_url).
```
Comment (Francois-Esquire) [inline docs/recipes/Reporting-Client-Errors.md:14]:
```suggestion
to the `reportingUrl` configured by the [`ONE_CLIENT_REPORTING_URL`](https://github.com/americanexpress/one-app/blob/main/docs/api/server/Environment-Variables.md#one_client_reporting_url) URL.
```

File: docs/recipes/Reporting-Client-Errors.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+<!--ONE-DOCS-HIDE end-->
+
+# Reporting Client Errors
+
+## `errorReporting` Duck
+
+`one-app` includes the [`errorReporting`](https://github.com/americanexpress/one-app-ducks#errorreporting-duck)
+duck from `one-app-ducks` which will log any errors reported by dispatching the [`addErrorToReport`](https://github.com/americanexpress/one-app-ducks#adderrortoreport) action. This provides your modules with a simple and efficient way to report errors from the client.
+
+When `addErrorToReport` is dispatched during SSR the `errorReporting` duck will log the reported error
```
Comment (Francois-Esquire) [inline docs/recipes/Reporting-Client-Errors.md:12]:
```suggestion
When `addErrorToReport` is dispatched during SSR, the `errorReporting` duck will log the reported error
```

**Review by Francois-Esquire — COMMENTED** [review]
Small nitpicks. Overall clear writing!

**Review by JamesSingleton — APPROVED** [review]

**Review by Francois-Esquire — APPROVED** [review]

**Review by infoxicator — APPROVED** [review]

**Pull request merged**